src/observability/publisher.go

```diff
@@ -58,32 +58,40 @@ func ProcessSystemSummary() {
 
 func getAggregatedSummaryMap(tempSummarizerMap map[types.SystemSummary]types.SysSummaryTimeCount) map[types.SystemSummary]types.SysSummaryTimeCount {
 
-	var fileArr []string
+	fileArr := make(map[string][]string)
+	fileArrTmp := make(map[string][]common.SysPath)
+	fileSumMap := make(map[types.SystemSummary]types.SysSummaryTimeCount)
+
 	log.Info().Msgf("Events before aggregation: [%v]", len(tempSummarizerMap))
 
-	fileSumMap := make(map[types.SystemSummary]types.SysSummaryTimeCount)
 	updatedSummarizerMap := make(map[types.SystemSummary]types.SysSummaryTimeCount)
```

Comment:
There is no need for this variable `updatedSummarizerMap`. Instead of copying non-file events to this map and then reassign `tempSummarizerMap`.  We could just delete the file events from `tempSummarizerMap`. We can avoid allocating additional memory.